### ISSUE TITLE
Nested classes in combination with TypedDict causes unexpected error report in mypy

### ISSUE THREAD

**Issue by svenzorgdoc** [python/mypy#8061]
Please provide more information to help us understand the issue:

* **Are you reporting a bug, or opening a feature request?**
This is a bug.
* **Please insert below the code you are checking with mypy,**
This file `works1.py` works okay:
```python
from typing_extensions import TypedDict

class A1:
    class Z(TypedDict):
        x: "B1.Y"

class B1:
    class Y(TypedDict):
        x: int
```

This file `works2.py` works okay:
```python
class A3:
    class :
        x: "B3.Q"

class B3:
    class Q:
        x: int
```

This file `works3.py` works okay:
```python
class A2:
    class Z:
        x: "B2.Z"

class B2:
    class Z:
        x: int

class A4:
    class Z:
        x: "B4.Z"

class B4:
    class Z:
        x: int
```

**However**, this file `does_not_work1.py` does **not** validate in mypy:
```python
from typing_extensions import TypedDict

class A2:
    class Z(TypedDict):
        x: "B2.Z"

class B2:
    class Z(TypedDict):
        x: int
```

This file `does_not_work2.py` does **not** validate either in mypy:
```python
from typing_extensions import TypedDict

class A2:
    class Z(TypedDict):
        x: "B2.Z"

class B2:
    class Z(TypedDict):
        x: int

# The code below DOES validate in works3.py, but is now suddenly broken due to
# the presence of the code above.

class A4:
    class Z:
        x: "B4.Z"

class B4:
    class Z:
        x: int
```

* **What is the actual behavior/output?**
The files `works1.py`, `works2.py` and `works3.py` all validate fine with mypy.
The files `does_not_work1.py` and `does_not_work2.py` do not validate with mypy and produce errors such as:
```
does_not_work2.py:5: error: Name 'B2.Z' is not defined
does_not_work2.py:16: error: Name 'B4.Z' is not defined
does_not_work1.py:5: error: Name 'B2.Z' is not defined
Found 3 errors in 2 files (checked 5 source files)
```
* **What is the behavior/output you expect?**
I expect all Python files above to validate with mypy.
* **What are the versions of mypy and Python you are using?**
mypy version 0.750, Python 3.7
* **Do you see the same issue after installing mypy from Git master?**
Yes, mypy 0.760+dev.f9955ae1101cd7d1b38c39e8fba77f1d74986a02 produces the same behavior.
* **What are the mypy flags you are using? (For example --strict-optional)**
No flags, but the same behavior is seen when using `--strict`.
* **If mypy crashed with a traceback, please paste the full traceback below.**
mypy does not crash.
* **Extra information:**
Using the same name for inner classes in various outer classes is normally not a problem, but as soon as `TypedDict` gets involved, things seem to go haywire.

**Comment by JukkaL** [comment]
It looks like nested TypedDict classes aren't reliably recognized as TypedDicts. Also the name of the nested TypedDict class may affect the behavior somehow. A simple workaround would be to move the TypedDict definitions to the module top level.

**Comment by svenzorgdoc** [comment]
> It looks like nested TypedDict classes aren't reliably recognized as TypedDicts. Also the name of the nested TypedDict class may affect the behavior somehow. A simple workaround would be to move the TypedDict definitions to the module top level.

The code above is a (very) simplified version of some Python code that is automatically generated from an API specification. We can't simply move the classes to the module level in this case.

What we've currently done is to rename the inner `TypedDict`s to make them unique among all other `TypedDict` names. That works for us at the moment.

I think `TypedDict` uses the class' `__name__` attribute instead of `__qualname__` or something similar (I am unfamiliar with the internals of mypy, I may be totally wrong in my thinking here). That would explain why it would work in non-nested environments because `__name__` is equal to `__qualname__` there.

What is also "funny" is that if you take the code from `does_not_work1.py`:
```python
from typing_extensions import TypedDict

class A2:
    class Z(TypedDict):
        x: "B2.Z"

class B2:
    class Z(TypedDict):
        x: int
```

and swap the two outer classes around, like this:
```python
from typing_extensions import TypedDict

class B2:
    class Z(TypedDict):
        x: int

class A2:
    class Z(TypedDict):
        x: "B2.Z"
```

then the problem is suddenly no longer present.

**Comment by 97littleleaf11** [comment]
I cannot reproduce the bugs on latest master branch. I think it has been fixed!
